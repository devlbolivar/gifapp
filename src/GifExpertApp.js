import React, { useState } from "react";

import { GifGrid } from "./components/GifGrid";
import { SearchBar } from "./components/SearchBar";
import { DarkModeBtn } from "./components/DarkModeBtn";
import { MainGif } from "./components/MainGif";
export const GifExpertApp = ({ defaultCategories = ["funny"] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div className="main__container">
      <div className="header">
        <h2 className="main__title left">GifExpertApp</h2>
        <MainGif />
        <div className="right"></div>
      </div>
      <div className="first__bar">
        <SearchBar setCategories={setCategories} />
        <DarkModeBtn />
      </div>

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </div>
  );
};
