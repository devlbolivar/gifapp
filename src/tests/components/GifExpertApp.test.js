import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../../GifExpertApp';

describe('Prueba sobre el componente <GifExpertApp />', () =>{
    test('Debe renderizar correctamente el componente', ()=>{
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorÃ­as', ()=>{
        const categories = ['funny', 'cool']; 
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);   
    });


});