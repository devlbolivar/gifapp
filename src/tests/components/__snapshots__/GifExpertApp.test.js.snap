// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba sobre el componente <GifExpertApp /> Debe de mostrar una lista de categor√≠as 1`] = `
<div
  className="main__container"
>
  <div
    className="header"
  >
    <h2
      class="main__title"
    >
      GifExpertApp
    </h2>
    <div
      className="main__gif"
    >
      <img
        alt="Random Img logo"
      />
    </div>
  </div>
  <div
    className="first__bar"
  >
    <SearchBar
      setCategories={[Function]}
    />
    <DarkModeBtn />
  </div>
  <hr />
  <ol>
    <GifGrid
      category="funny"
      key="funny"
    />
    <GifGrid
      category="cool"
      key="cool"
    />
  </ol>
</div>
`;

exports[`Prueba sobre el componente <GifExpertApp /> Debe renderizar correctamente el componente 1`] = `
<div
  className="main__container"
>
  <div
    className="header"
  >
    <h2
      class="main__title"
    >
      GifExpertApp
    </h2>
    <div
      className="main__gif"
    >
      <img
        alt="Random Img logo"
      />
    </div>
  </div>
  <div
    className="first__bar"
  >
    <SearchBar
      setCategories={[Function]}
    />
    <DarkModeBtn />
  </div>
  <hr />
  <ol />
</div>
`;
